frame/transaction-storage/README.md

```diff
@@ -1,8 +1,63 @@
 # Transaction Storage Pallet
 
 Indexes transactions and manages storage proofs.
-# Transaction Storage Pallet
 
-Indexes transactions and manages storage proofs.
+Allows storing arbitrary data on the shain. Data is automaticcaly removed after `StoragePeriod` blocks, unless the storage is renewed.
+Validators must submit proof of storing a random chunk of data for block `N - StoragePeriod` when producing block `N`.
+
+# Runing a chain
+
+The following describes how to setup a new storage chain.
+
+Start with generating a chain spec.
+```
+cargo run --release -- build-spec --chain=local > sc_init.json
+```
+Edit the json chain spec file to customise the chain. The storage chain genesis params are configured in the `transactionStorage` section.
+Note that `storagePeriod` is specified in blocks and changing it also requires code changes at the moment.
+
+Build a raw spec from the init spec.
+```
+cargo run --release build-spec --chain=sc_init.json --raw > sc.json
+```
+Run a few validator nodes.
+
+```
+cargo run --release -- --chain=sc.json -d /tmp/alice --storage-chain --keep-blocks=100800 --ipfs-server --validator --alice
+cargo run --release -- --chain=sc.json -d /tmp/bob --storage-chain --keep-blocks=100800 --ipfs-server --validator --bob
+```
+
+`--storage-chain` enables transaction indexing.
+`--keep-blocks=100800` enables block pruning. The value here should be greater or equal than the storage period.
+`--ipfs-server` enables serving stored content over IPFS.
+
+Once the network is started, any other joining nodes need to sync with `--sync=fast`. Regular sync will fail because block pruning removes old blocks. The chain does noot keep full block history.
```

Comment:
```suggestion
Once the network is started, any other joining nodes need to sync with `--sync=fast`. Regular sync will fail because block pruning removes old blocks. The chain does not keep full block history.
```